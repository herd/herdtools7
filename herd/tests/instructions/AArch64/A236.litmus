AArch64 236

(* Tests MVN Register instruction *)
(* Remember inverting 00 -> 11 which is 0 to -1 in 2s complement *)
{ int64_t 0:X1 = 0; int64_t 0:X2 = 1;
  int64_t 0:X0; int64_t 0:X3; }

P0;
  MVN X0, X1;
  MVN X3, X2;

forall (0:X0 = -1 /\ 0:X3 = -2)
