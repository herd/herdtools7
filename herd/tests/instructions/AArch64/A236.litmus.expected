Test 236 Required
States 1
0:X0=-1; 0:X3=-2;
Ok
Witnesses
Positive: 1 Negative: 0
Condition forall (0:X0=-1 /\ 0:X3=-2)
Observation 236 Always 1 0
Hash=8205eb706c7f2a58368a8bdc3b855c37

